src/test/tasks/operator_task_test.cpp

```diff
@@ -95,5 +105,16 @@ TEST_F(OperatorTaskTest, MakeDiamondShape) {
 
   std::vector<std::shared_ptr<AbstractTask>> expected_successors_4{};
   EXPECT_EQ(tasks[4]->successors(), expected_successors_4);
+
+  for (auto& task : tasks) {
```

Comment:
this could just be a one liner CurrentScheduler::schedule_and_wait_for_tasks()

Reply:
That's what I thought, but if I understand it correctly, the idea is to keep the tests for the OperatorTask free from any scheduler things.

Reply:
Imo Tasks are part of the Scheduler component, so calling the scheduler here wouldn't be a problem with me. But it's a minor thing anyway, do as you wish.